src/Storages/Streaming/StreamingStoreSourceBase.cpp

```diff
@@ -161,13 +171,20 @@ Chunk StreamingStoreSourceBase::generate()
 
 /// It basically initiate a checkpoint
 /// Since the checkpoint method is called in a different thread (CheckpointCoordinator)
-/// We nee make sure it is thread safe
+/// We need make sure it is thread safe
 void StreamingStoreSourceBase::checkpoint(CheckpointContextPtr ckpt_ctx_)
 {
     /// We assume the previous ckpt is already done
     /// Use std::atomic<std::shared_ptr<CheckpointContext>>
-    assert(!ckpt_ctx.load(std::memory_order_relaxed));
-    ckpt_ctx = new CheckpointContext(*ckpt_ctx_);
+    assert(!has_ckpt_request);
+    CheckpointContextPtr new_ckpt = std::make_shared<CheckpointContext>(*ckpt_ctx_);
```

Comment:
Do we need make a new_ckpt ?

Reply:
i got many many error about the ckpt file path duplicate(i assume this is due to the shallow copy)
work with deep copy everything well.

Reply:
Could you elaborate the error ? When it happened and what's the exact error message ?